src/utils/wallet.js

```diff
@@ -70,26 +74,66 @@ export function catchTransactionError(e: Object, type: string, tx: Object) {
 }
 
 // handle eth_signTransaction
-export async function signTransaction(trx: Object, wallet: Object): Promise<string> {
+export function signTransaction(trx: Object, wallet: Object): Promise<string> {
   wallet.provider = providers.getDefaultProvider(NETWORK_PROVIDER);
   if (trx && trx.from) {
     delete trx.from;
   }
-  const result = await wallet.sign(trx);
-  return result;
+  return wallet.sign(trx);
 }
 
 // handle eth_sign
-export async function signMessage(message: any, wallet: Object): Promise<string> {
+export function signMessage(message: any, wallet: Object): string {
   wallet.provider = providers.getDefaultProvider(NETWORK_PROVIDER);
   // TODO: this method needs to be replaced when ethers.js is migrated to v4.0
-  const result = ethSign(message, wallet.privateKey);
-  return result;
+  return ethSign(message, wallet.privateKey);
 }
 
 // handle personal_sign
-export async function signPersonalMessage(message: string, wallet: Object): Promise<string> {
+export function signPersonalMessage(message: string, wallet: Object): Promise<string> {
   wallet.provider = providers.getDefaultProvider(NETWORK_PROVIDER);
-  const result = await wallet.signMessage(isHexString(message) ? ethers.utils.arrayify(message) : message);
-  return result;
+  return wallet.signMessage(isHexString(message) ? ethers.utils.arrayify(message) : message);
+}
+
+// we use basic AsyncStorage implementation just to prevent backup being stored in same manner
+export async function getWalletFromStorage(dispatch: Dispatch, appSettings: Object) {
+  let { wallet = {} } = await storage.get('wallet');
+  const walletBackup = await AsyncStorage.getItem(WALLET_STORAGE_BACKUP_KEY);
+  const isWalletEmpty = !wallet || !Object.keys(wallet).length;
+  // wallet timestamp missing causes welcome screen
+  let walletTimestamp = appSettings.wallet;
+  if (isWalletEmpty && walletBackup) {
+    console.log('RESTORING WALLET FROM BACKUP');
+    // restore wallet to storage
+    wallet = JSON.parse(walletBackup);
```

Comment:
try..catch ?

Reply:
Why? Nothing to put in catch case as we just check if there was no wallet and backup exists and then restore it. I. e. if we put `wallet = JSON.parse(walletBackup)` and save in try then catch will still remain empty, or you're thinking of something else?